### PR TITLE
Create Option class for working with options

### PR DESCRIPTION
**Pull request by davydovanton**
I found [this TODO](https://github.com/lotus/model/blob/master/lib/lotus/model/configuration.rb#L265-L268) and I decided to do it.

### CODE REVIEW

File: lib/lotus/utils/options.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require 'lotus/utils/hash'
+
+module Lotus
+  module Utils
+    # This class provide methods for options.
+    #
+    # @since x.x.x
+    class Options < Lotus::Utils::Hash
+      # Check options on inclided key or keys
```
Comment (joneslee85) [inline lib/lotus/utils/options.rb]: s/inclided/included

File: lib/lotus/utils/options.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'lotus/utils/hash'
+
+module Lotus
+  module Utils
+    # This class provide methods for options.
```
Comment (joneslee85) [inline lib/lotus/utils/options.rb]: s/provide/provides

**Comment by davydovanton** [comment]
@joneslee85 thanks for catch! fixed.

File: test/options_test.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          options.check_options!(:foo)
+        }.must_be_silent
+      end
+    end
+
+    describe 'when options not contain a keys' do
+      it 'raises exeption' do
+        -> {
+          options.check_options!(:baz, :foobar)
+        }.must_raise(ArgumentError)
```
Comment (joneslee85) [inline test/options_test.rb]: can you please also check the exception message?

File: test/options_test.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+require 'lotus/utils/options'
+
+describe Lotus::Utils::Options do
+  let(:options) { Lotus::Utils::Options.new(foo: :bar) }
+
+  describe '#check_options!' do
+    describe 'when options contain a keys' do
```
Comment (joneslee85) [inline test/options_test.rb]: keys is a plural, I think you should remove 'a' or the 's'

File: test/options_test.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      it 'not raises exeption' do
```
Comment (joneslee85) [inline test/options_test.rb]: s/not/does not

Comment (joneslee85) [inline test/options_test.rb]: s/raises/raise

**Comment by davydovanton** [comment]
@joneslee85 I added test for error message. Fixed grammar :tada:

**Comment by joneslee85** [comment]
@davydovanton thanks for the PR. Ruby 2.0 will be EOL on Feb 16th 2016, that also means Lotus will be able to use `kwargs`. Taking into the fact that it is pretty close now, I am questioning if we still need this util method.

**Comment by joneslee85** [comment]
cc @jodosha

**Comment by davydovanton** [comment]
@joneslee85 interesting question. So I see two reasons why these changes are helpful:
1. These changes help to avoid duplication of code now
2. We can change this code to `kwargs` method faster in future

**Comment by jodosha** [comment]
@davydovanton Yes, @joneslee85 is right. For that specific use case linked by you, it isn't needed anymore. Can you please check the other similar cases? Thank you!

**Comment by davydovanton** [comment]
so, yes I think this class will be helpful only for old ruby version without kwargs.

**Comment by jodosha** [comment]
@davydovanton Okay, thanks for this PR in the first place and thanks for checking the other use cases. :smile: Soon we'll support only Ruby versions that implements kwargs. Fro this reason I'm closing this. Thank you again :dog:

**Comment by davydovanton** [comment]
No problem :)
I happy to help :)

**Comment by jodosha** [comment]
@davydovanton :)

**Pull request closed without merging**